tensorboard/loader_test.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import functools
+import os
+
+import tensorflow as tf
+
+from tensorboard import loader
+from tensorboard import test_util
+from tensorboard import util
+
+
+class LoaderTestCase(tf.test.TestCase):
+  def __init__(self, *args, **kwargs):
+    super(LoaderTestCase, self).__init__(*args, **kwargs)
+    self.clock = test_util.FakeClock()
+
+  def _save_string(self, name, data):
+    """Writes new file to temp directory.
+
+    :type name: str
+    :type data: str
+    """
+    path = os.path.join(self.get_temp_dir(), name)
+    with open(path, 'wb') as writer:
+      writer.write(tf.compat.as_bytes(data))
+    return path
+
+  def _save_records(self, name, records):
+    """Writes new record file to temp directory.
+
+    :type name: str
+    :type records: list[str]
+    :rtype: str
+    """
+    path = os.path.join(self.get_temp_dir(), name)
+    with RecordWriter(path) as writer:
+      for record in records:
+        writer.write(record)
+    return path
+
+
+class BufferedRecordReaderTest(LoaderTestCase):
```

Comment:
Offset is now tested. I tried testing buffering, but realized it would be highly difficult because tf.test.mock.Mock doesn't support threads :'(